src/components/Form/rfc.md

```diff
@@ -0,0 +1,467 @@
+| showSubmitButton               | Set to `true` to display the submit button when `onSubmit` is set.                                                                                                                         |
+| submitButton                   | Boolean determining whether to show the submit button.                                                                                                                                     |
+| valueKey                       | Specifies the required value for each field.                                                                                                                                               |
+
+### Key Functionalities Enabled by These Props
+
+1. **Simplified Rendering of Common Components**  
+   These include features such as a **submit button**, **custom actions**, **wizard elements**, and layout management (e.g., **column alignment**, button positioning, or forms within modals).
+
+2. **Unified Handlers and Behaviors for Input Interactions**  
+   Props like `onChange` and `onBlur` ensure consistent behavior for inputs, while configuration options such as `cleanFieldsOnSubmit` apply to all fields, and `groups` allow targeting specific subsets of fields.
+
+3. **Unified Initialization of All Inputs via a Single `fields` Prop**  
+   The `fields` prop enables seamless initialization of input components, while still allowing for more advanced input logic when needed.
+
+### Limitations
+
+- **Layout Customization:**  
+  The layout is **not highly customizable**, as it relies solely on basic configuration props at the parent component level. This makes it **difficult to insert intermediate components or wrappers** between the parent form and individual inputs.
+
+- **Custom Input Implementation:**  
+  The `fields` prop is based on **getInputDecorator**, which is **deprecated in Ant Design 5**. A new strategy is required for creating custom input components outside the design system.
+
+- **Form State Management and Validation:**  
+  Form state management is implemented in the parent component using the parent form’s state, which is currently implemented with a **class component**. This approach is now deprecated in more recent versions of React.
+
+## Proposal/Design
+
+The purpose of this analysis is to provide guidelines and an implementation strategy addressing the limitations highlighted above.
+
+### Layout Customization
+
+Given the limitations of configuring form fields via the `fields` prop, the proposal is to **deprecate this property** and instead define form items directly using React elements. The form items would be configured through the props supported by the component.
+
+This approach allows for a seamless coexistence of form items and custom elements within the DOM:
+
+```jsx
+<Form>
+  <FormItem name="input-name" {...fieldProps} />
```

Comment:
How will this be imported in the user codebase?

Do you expect to use

```js
import {Form, FormItem} from 'design-system'

function Comp({}){
  return <Form ... ><FormItem ...></Form>
}
```

of

```js
import {Form} from 'design-system'

function Comp({}){
  return <Form ... ><Form.Item ...></Form>
}
```

I'd prefer the last one (which is also similar to other subcomponent exports that have been done in the lib)